server/src/main/java/com/google/fhir/gateway/BearerAuthorizationInterceptor.java

```diff
@@ -192,12 +214,42 @@ public boolean authorizeRequest(RequestDetails requestDetails) {
       } else {
         reader = HttpUtil.readerFromEntity(entity);
       }
+
+      if (isEventLoggingEnabled) {
+        Reference agentUserWho = outcome.getUserWho(requestDetailsReader);
+        if (agentUserWho != null) {
+
+          StringWriter responseStringWriter = new StringWriter();
+          reader.transferTo(responseStringWriter);
```

Comment:
How does this work in the case that `reader = HttpUtil.readerFromEntity(entity)` (line 215 above)? I mean, IIUC, we are reading all characters from the input stream and then in the `replaceAndCopyResponse` below, we won't be able to re-read the `reader`, hence not fixing the URLs. Is my understanding correct? If yes, can you please fix this and also add a unit-test that covers this case?

Reply:
I think this is fine because of the assignment of `reader` on line 241 below; resolving.